### ISSUE TITLE
feature_interaction_constraint like xgboost

### ISSUE THREAD

**Issue by marquisthunder** [google/yggdrasil-decision-forests#129]
Dear YDF team,
    As it is possible to specify feature interaction constraint[1] in xgboost to use domain knowledge to improve the model, is it possible to  add this feature in ydf? It would be great when combined with oblique split as sometimes the oblique split condition is difficult to use in model explanation.

Referecense:
1. https://xgboost.readthedocs.io/en/stable/tutorials/feature_interaction_constraint.html

**Comment by rstz** [comment]
I agree that this is a nice feature to have, we'll consider it for our roadmap!
